extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php

```diff
@@ -6,25 +6,30 @@ public function __construct() {
 		parent::__construct('CreateNewWiki', 'createnewwiki');
 	}
 
+	/**
+	 * @param string $par
+	 * @throws ErrorPageError
+	 */
 	public function execute( $par ) {
-		global $wgUser, $wgOut;
 		wfProfileIn( __METHOD__ );
+		$out = $this->getOutput();
+
+		$this->checkPermissions();
 
 		if ( wfReadOnly() ) {
-			$wgOut->readOnlyPage();
+			$out->readOnlyPage();
 			wfProfileOut(__METHOD__);
 			return;
 		}
 
-		if (!$wgUser->isAllowed('createnewwiki')) {
-			$this->displayRestrictionError();
-			wfProfileOut( __METHOD__ );
-			return;
+		// SUS-1182: CreateNewWiki should check for valid email before progressing to the second step
+		if ( !$this->getUser()->isEmailConfirmed() ) {
+			throw new ErrorPageError( 'cnw-error-unconfirmed-email-header', 'cnw-error-unconfirmed-email' );
 		}
 
-		$wgOut->setPageTitle(wfMsg('cnw-title'));
+		$out->setPageTitle(wfMsg('cnw-title'));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)